### PR TITLE
Handling failed AJAX requests

### PR DESCRIPTION
**Pull request by kppf**
Signed-off-by: Kushagra Pandey kushagra4296@gmail.com
- If an ajax request fails, the [Loading...] notification is replaced by [Request Failed] and is dismissed after default time (5000ms)
- Figured out that there was a bug which prevented any AJAX request once any AJAX request has failed. So corrected that too by making AJAX.active = false; and AJAX.xhr = null;
- Also handles the cases when the returned data is not JSON, which is a fail for the client. Shows [Request Failed] for this too.

Update from previous pull request: No more blocking the mechanism of aborting a new call.

### CODE REVIEW

**Comment by zixtor** [comment]
Now, on two subsequent link clicks, it allows the second request to get fired without actually aborting the first one, so now both requests show their effect, which is not the behavior as before this patch.

File: js/console.js [hunk 1]
```diff
@@ -184,6 +184,12 @@ var PMA_console = {
                     PMA_console.ajaxCallback(data);
                 } catch (e) {
                     console.log("Invalid JSON!" + e.message);
+                    if(AJAX.cancellingRequest++ > 0) {
```
Comment (zixtor) [inline js/console.js]: Here we can avail AJAX.xhr.statusText..?
Modifying above condition to (AJAX.xhr.status === 0 && AJAX.xhr.statusText !== 'abort') seems to work..

File: js/console.js [hunk 2]
```diff
@@ -184,6 +184,12 @@ var PMA_console = {
...
+                        PMA_ajaxShowMessage($('<div />',{class:'error',html:PMA_messages.strRequestFailed}));
```
Comment (zixtor) [inline js/console.js]: Additionally, we can use statusText here also, to give more information to the user, like 'timeout' or 'error'.

File: js/console.js (hunk 1 above)
Comment (kppf) [inline js/console.js]: Thank you. I was confused with the call flow. That's why did the wrong thing. Thank you. Sent another pull request.
